spec/integration/auto_encryption_spec.rb

```diff
@@ -440,6 +442,44 @@
         it_behaves_like 'it obeys bypass_auto_encryption option'
       end
     end
+
+    context 'encrypting using key alt name' do
+      include_context 'schema map in client options'
+
+      let(:query) { { ssn: ssn, altname: key_alt_name } }
+
+      context 'with AWS KMS provider' do
+        include_context 'with AWS kms_providers and key alt names'
+        it 'encrypts the ssn field' do
+          expect(result).to be_ok
+          expect(result.inserted_ids.length).to eq(1)
+
+          id = result.inserted_ids.first
+
+          document = client[:users].find(_id: id).first
+          document.should_not be_nil
+          # Auto-encryption with key alt names only works with random encryption,
+          # so it will not generate the same result on every test run.
+          expect(document['ssn']).to be_a_kind_of(BSON::Binary)
+        end
+      end
+
+      context 'with local KMS provider' do
+        include_context 'with local kms_providers and key alt names'
+        it 'encrypts the ssn field' do
+          expect(result).to be_ok
+          expect(result.inserted_ids.length).to eq(1)
+
+          id = result.inserted_ids.first
+
+          document = client[:users].find(_id: id).first
+          document.should_not be_nil
+          # Auto-encryption with key alt names only works with random encryption,
+          # so it will not generate the same result on every test run.
+          expect(document['ssn']).to be_a_kind_of(BSON::Binary)
```

Comment:
I think this is an example of a situation where it makes sense to write an "is ciphertext" matcher, which would check the type and the bson subtype.

Reply:
Yeah I like that!